src/App.js

```diff
@@ -1,122 +1,99 @@
 import "./App.css";
-import { BrowserRouter, Route, Routes, useNavigate } from "react-router-dom";
+import {
+  BrowserRouter,
+  Route,
+  Routes,
+  useParams,
+  Link,
+  Outlet,
+  useLocation
+} from "react-router-dom";
+import { useState, useEffect } from "react";
 import FourOhFour from "./components/PageNotFound/PageNotFound";
-import { useState } from "react";
-import "../src/App.css";
-import axios from "axios";
+import PokemonDetailsPage from "./pages/PokemonDetailsPage/PokemonDetailsPage.jsx";
 
 function PokeApp() {
-    const navigate = useNavigate();
-    const [pokemonName, setPokemonName] = useState("");
-    const [isChosen, setIsChosen] = useState(false);
-    const [pokemonInfo, setPokemonInfo] = useState({
-        name: "",
-        species: "",
-        imageUrl: "",
-        hp: "",
-        attack: "",
-        defense: "",
-        type: "",
-    });
-
-    const searchPokemon = () => {
-        const lowerCasePokemonName = pokemonName.toLowerCase();
-        axios
-            .get(`https://pokeapi.co/api/v2/pokemon/${lowerCasePokemonName}`)
-            .then((response) => {
-                setPokemonInfo({
-                    name: pokemonName,
-                    species: response.data.species.name,
-                    imageUrl: response.data.sprites.front_default,
-                    hp: response.data.stats[0].base_stat,
-                    attack: response.data.stats[1].base_stat,
-                    defense: response.data.stats[2].base_stat,
-                    type: response.data.types[0].type.name,
-                });
-            })
-            .catch((error) => {
-                if (error.response) {
-                    if (error.response.status === 404) {
-                        navigate("/404");
-                        return;
-                    }
-                }
-            });
-        setIsChosen(true);
+  const location = useLocation();
+  const [isChosen, setIsChosen] = useState(false)
+  const { pokemonName: urlPokemonName } = useParams();
+  const [pokemonName, setPokemonName] = useState(urlPokemonName || "");
+ 
+  useEffect(() => {
+    const handleKeyDown = (event) => {
+      if (event.key === "Enter") {
+        document.getElementById("search-button").click();
+      }
```

Comment:
HTML forms already have a mechanism for handling submit. 

If you wrap your input below in a `<form>` it will expose an `onSubmit` method that is called when the user hits the Enter key. 

It's much more accessible to users and removes the need for you to manage the `keyDown` event yourself.

i.e

```jsx
const handleSubmit = (event) => {
	event.preventDefault() // This stops the page refreshing when you submit the form
	/// ... all your other search logic
}


<form onSubmit={handleSubmit}>
	</input ..../>
</form>
```

Reply:
Oh man this is so much better. Despite all the HTML and form lessons I've been through, none ever mention that.

Reply:
I would avoid resolving other people's comments in PR's btw, especially if you're yet to push the change that resolves it. 

From my perspective I can't see your comment unless I go looking for the hidden comments and I can't see any changes to the code. So it looks like you just deleted my comment and ignored it.

Reply:
Ah yes, okay. Good etiquette/practice lesson for me.